message_test.go

```diff
@@ -807,6 +808,219 @@ func TestPackUnpack(t *testing.T) {
+		require.Equal(t, "00", s)
+
+		data := &TestISOShortData{}
+		require.NoError(t, message.Unmarshal(data))
+
+		assert.Equal(t, "4276555555555555", data.F2.Value())
+	})
+
+	t.Run("Unpack data field error on subfield in nested spec returns both numbers and partial message", func(t *testing.T) {
+		type TestDateAndTimeData struct {
+			F1 *field.String
+			F2 *field.Numeric
+		}
+
+		type TestISOF3MixedData struct {
+			F1 *field.String
+			F2 *TestDateAndTimeData
+			F3 *field.String
+		}
+
+		type TestISOShortData struct {
+			F2 *field.String
+			F3 *TestISOF3MixedData
+		}
+
+		shortSpec := &MessageSpec{
+			Fields: map[int]field.Field{
+				0: field.NewString(&field.Spec{
+					Length:      4,
+					Description: "Message Type Indicator",
+					Enc:         encoding.ASCII,
+					Pref:        prefix.ASCII.Fixed,
+				}),
+				1: field.NewBitmap(&field.Spec{
+					Description: "Bitmap",
+					Enc:         encoding.Binary,
+					Pref:        prefix.ASCII.Fixed,
+				}),
+				2: field.NewString(&field.Spec{
+					Length:      19,
+					Description: "Primary Account Number",
+					Enc:         encoding.ASCII,
+					Pref:        prefix.ASCII.LL,
+				}),
+				3: field.NewComposite(&field.Spec{
+					Length:      14,
+					Description: "Processing Code",
+					Pref:        prefix.ASCII.LL,
+					Tag: &field.TagSpec{
+						Sort: sort.StringsByInt,
+					},
+					Subfields: map[string]field.Field{
+						"1": field.NewString(&field.Spec{
+							Length:      2,
+							Description: "Transaction Type",
+							Enc:         encoding.ASCII,
+							Pref:        prefix.ASCII.Fixed,
+						}),
+						"2": field.NewComposite(&field.Spec{
+							Length:      10,
+							Description: "Authorization System Advice Date and Time",
+							Pref:        prefix.ASCII.Fixed,
+							Tag: &field.TagSpec{
+								Sort: sort.StringsByInt,
+							},
+							Subfields: map[string]field.Field{
+								"1": field.NewString(&field.Spec{
+									Length:      4,
+									Description: "Date",
+									Enc:         encoding.ASCII,
+									Pref:        prefix.ASCII.Fixed,
+								}),
+								"2": field.NewNumeric(&field.Spec{
+									Length:      6,
+									Description: "Time",
+									Enc:         encoding.ASCII,
+									Pref:        prefix.ASCII.Fixed,
+								}),
+							},
+						}),
+						"3": field.NewString(&field.Spec{
+							Length:      2,
+							Description: "To Account",
+							Enc:         encoding.ASCII,
+							Pref:        prefix.ASCII.Fixed,
+						}),
+					},
+				}),
+			},
+		}
+		message := NewMessage(shortSpec)
+		message.MTI("0100")
+
+		// Field F3SF2SF2 is numeric type but has a letter in, which will cause an unpack error
+		rawMsg := []byte{0x30, 0x31, 0x30, 0x30, // MTI F0
+			0x60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, // Bitmap F1
+			0x31, 0x36, // Tag for F2
+			0x34, 0x32, 0x37, 0x36, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, 0x35, //F2
+			0x31, 0x34, 0x30, 0x30, 0x31, 0x31, 0x31, //F3 SF1
```

Comment:
I think `0x31, 0x34` is the length tag, then `0x30, 0x30` is F3 SF1, then `0x31, 0x31, 0x31` is the start of F3 SF1